monai/utils/misc.py

```diff
@@ -352,6 +356,95 @@ def copy_to_device(
     return obj
 
 
+def get_dist_device():
+    """
+    Get the expected target device in the distributed data parallel.
+    For NCCL backend, return GPU device of current process.
+    For GLOO backend, return CPU.
+    For any other backends, return None as the default, tensor.to(None) will not change the device.
+
+    """
+    if dist.is_initialized():
+        backend = dist.get_backend()
+        if backend == "nccl" and torch.cuda.is_available():
+            return torch.device(f"cuda:{torch.cuda.current_device()}")
+        elif backend == "gloo":
+            return torch.device("cpu")
+    return None
+
+
+def evenly_divisible_all_gather(data: torch.Tensor, concat: bool = False):
```

Comment:
if `concat` default to True, will we have better backward compatibility?

Reply:
I am OK to set it default to True, actually, I am not very sure the `False` value. Let me update it.

Thanks.